roles/network_plugin/calico/templates/cni-calico.conflist.j2

```diff
@@ -6,7 +6,7 @@
     {% if cloud_provider is defined %}
       "nodename": "{{ calico_kubelet_name.stdout }}",
     {% else %}
-      "nodename": "{{ ansible_hostname }}",
+      "nodename": "{{ inventory_hostname }}",
```

Comment:
If ansible_hostname != inventory_hostname, calico will break here:
https://github.com/kubernetes-incubator/kubespray/blob/master/roles/kubernetes/node/defaults/main.yml#L87

CNI will set node name to one value and kubelet will set another, which is then used here for calico daemonset:
https://github.com/kubernetes-incubator/kubespray/blob/master/roles/network_plugin/calico/templates/calico-node.yml.j2#L115-L118

So, we should update the kubelet node name to match here, but I'm pretty sure this will break existing installs.

So I should ask why is there a strong case to not use the same hostname in inventory as you use for hosts?

Reply:
Ah, I overrode the other in inventory with ```kube_override_hostname: "{{ inventory_hostname }}"``` to match.  

From what I found, the SSL certs get generated with inventory_hostname which we want to be fqdn, but the ansible hostname is only the shortname.  So the defaults for the non-cloudprovider case would not allow k8s to start without fqdn in there, and then the calico hardcode to ansible_hostname would not allow calico to work/match the fqdn.

I think it makes sense to change the default for k8s hostname override to inventory_hostname as well in https://github.com/kubernetes-incubator/kubespray/blob/master/roles/kubernetes/node/defaults/main.yml#L87. If you agree I will add it into this PR.

Reply:
Let's see how much this breaks upgrades :) Can you make that change? We have a CI case for upgrades

Reply:
Think something is wonky with report status, but the pipeline passed if you click details.  Ready to merge or want more changes?